source/backend/api/Services/LeaseService.cs

```diff
@@ -184,6 +185,11 @@ public PimsLease Add(PimsLease lease, IEnumerable<UserOverrideCode> userOverride
             var pimsUser = _userRepository.GetByKeycloakUserId(_user.GetUserKey());
             pimsUser.ThrowInvalidAccessToLeaseFile(lease.RegionCode);
 
+            if (lease.PimsPropertyLeases.Any(x => x.Property.IsRetired.HasValue && x.Property.IsRetired.Value))
```

Comment:
It seems like your pattern here is to add this logic to both the service and also to the repository, but not in the case of leases?

Reply:
I don't think you can update the property on a lease?

Reply:
Updated